From c2b14150d81dd1b3afb0dcb93f1ed1584839df93 Mon Sep 17 00:00:00 2001
From: Lei Qian <lei.qian@amlogic.com>
Date: Fri, 30 Nov 2018 22:40:21 +0800
Subject: [PATCH] AudioPolicyManager: set default media volume if
 volumeMaxIndex is 1 [1/1]

PD#SWPL-2152

Problem:
bootvideo has no sound when first bootup

Solution:
set default media volume if volumeMaxIndex is 1

Verify:
verify by p321
Signed-off-by: Lei Qian <lei.qian@amlogic.com>

Change-Id: I31124c18a78ffeb2702a4fafefb700197cbd6ee1
---
 services/audiopolicy/managerdefault/AudioPolicyManager.cpp | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/services/audiopolicy/managerdefault/AudioPolicyManager.cpp b/services/audiopolicy/managerdefault/AudioPolicyManager.cpp
index 34f5a23..cda8741 100644
--- a/services/audiopolicy/managerdefault/AudioPolicyManager.cpp
+++ b/services/audiopolicy/managerdefault/AudioPolicyManager.cpp
@@ -5660,6 +5660,17 @@ status_t AudioPolicyManager::checkAndSetVolume(audio_stream_type_t stream,
             int volumeIndex = mVolumeCurves->getVolumeIndex(AUDIO_STREAM_MUSIC, device);
             int volumeMaxIndex = mVolumeCurves->getVolumeIndexMax(AUDIO_STREAM_MUSIC);
             float mediaVolume = (float) volumeIndex / (float) volumeMaxIndex;
+
+            if (volumeMaxIndex == 1) {
+                char defaultMediaVolume[PROPERTY_VALUE_MAX];
+                char volumeStep[PROPERTY_VALUE_MAX];
+                if (property_get("ro.config.media_vol_default", defaultMediaVolume, NULL)
+                        && property_get("ro.config.media_vol_steps", volumeStep, NULL)) {
+                   mediaVolume = ((float)atoi(defaultMediaVolume)) / ((float)atoi(volumeStep));
+                } else {
+                   mediaVolume = 0.25f;
+                }
+            }
             outputDesc->updateGain(stream, device, mediaVolume);
         }
     }
-- 
1.9.1

